src/librbd/AioRequest.cc

```diff
@@ -187,7 +288,34 @@ namespace librbd {
 				 << m_object_image_extents << dendl;
 
 	  m_state = LIBRBD_AIO_WRITE_COPYUP;
-	  read_from_parent(m_object_image_extents);
+
+          if (is_copy_on_read(m_ictx, m_snap_id)) {
+	    m_ictx->copyup_list_lock.Lock();
```

Comment:
It looks like this should just be a Mutex::Locker, since read_from_parent() can delete the CopyupRequest triggering copyup_list manipulation that should be protected in the destructor.

In the second branch, append_request() also modifies internal state that should be protected by this lock